app/components/e-heading.vue

```diff
@@ -0,0 +1,181 @@
+<!-- This component has no <template> because of dynamic root element -->
+
+<script>
+  export default {
+    name: 'e-heading',
+    // components: {},
+    // mixins: [],
+
+    props: {
+      /**
+       * Defines the color of the heading (h1-h6)
+       */
+      color: {
+        type: String,
+        default: 'blue',
+        validator(value) {
+          return [
+            'blue',
+            'green',
+            'gray',
+          ].includes(value);
+        },
+      },
+
+      /**
+       * Defines the tagname of the heading (h1-h6)
+       */
+      tagName: {
+        type: String,
+        required: true,
+        validator(value) {
+          return [
+            'h1',
+            'h2',
+            'h3',
+            'h4',
+            'h5',
+            'h6',
+          ].includes(value);
+        },
+      },
+
+      /**
+       * Defines if the heading should have an underline
+       */
+      underline: {
+        type: Boolean,
+        default: true,
+      },
+
+      /**
+       * Defines the font weight
+       */
+      weight: {
+        type: String,
+        default: 'normal',
+        validator(value) {
+          return [
+            'normal',
+            'semibold',
+            'bold',
+          ].includes(value);
+        },
+      },
+    },
+
+    // computed: {},
+    // watch: {},
+
+    // beforeCreate() {},
+    // created() {},
+    // beforeMount() {},
+    // mounted() {},
+    // beforeUpdate() {},
+    // updated() {},
+    // activated() {},
+    // deactivated() {},
+    // beforeDestroy() {},
+    // destroyed() {},
+
+    // methods: {},
+
+    /**
+     * Creates a heading element
+     *
+     * @param   {function}    createElement   Vue helper
+     *
+     * @returns {*}
+     */
+    render(createElement) {
+      const element = this.$props.tagName;
+      const attributes = {
+        class: this.b({
+          color: this.$props.color,
+          underline: this.$props.underline,
+          weight: this.$props.weight,
+          tagname: this.$props.tagName,
+        }),
+        attrs: {
+          ...this.$attrs,
+        },
+      };
+
+      return createElement(
+        element,
+        attributes,
+        this.$slots.default,
+      );
+    },
+  };
+</script>
+
+<style lang="scss">
+  .e-heading {
+    font-family: $font-family--primary;
+    font-weight: normal;
+    color: $color-secondary--1;
+    letter-spacing: 2.03px;
+    line-height: 23px;
```

Comment:
- [ ] Please don't use units for `line-height`. Use a multiplicator number. If nothing defined use `1.25` which is a good default. This will also result in customized `line-height` for each heading size.

Reply:
:+1: